cmake/common.cmake

```diff
@@ -189,70 +199,206 @@ Unit Test Specification
+    set(COMPONENTS_INFO ${COMPONENTS_INFO} PARENT_SCOPE)
+endmacro()
+
+macro(_spl_create_docs_target)
+    set(_docs_out_dir ${CMAKE_CURRENT_BINARY_DIR}/docs)
+    set(_docs_html_out_dir ${_docs_out_dir}/html)
+    set(_docs_config_json ${_docs_out_dir}/config.json)
+    
+    # create the config.json file. This is exported as SPHINX_BUILD_CONFIGURATION_FILE env variable
+    list(JOIN COMPONENTS_INFO "," _components_info_json)
+    set(_components_info_json "[${_components_info_json}]")
+    list(JOIN COMPONENTS_SPHINX_INCLUDE_PATTERNS "\",\"" _components_sphinx_include_patterns_json)
+    set(_components_sphinx_include_patterns_json "[\"${_components_sphinx_include_patterns_json}\"]")
+    file(WRITE ${_docs_config_json} "{
+    \"target\": \"docs\",
+    \"include_patterns\": ${_components_sphinx_include_patterns_json},
+    \"components_info\": ${_components_info_json}
+}")
+
+    # add the generated files as dependency to cmake configure step
+    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${_docs_config_json})
+    add_custom_target(
+        docs
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${_docs_out_dir}
+        COMMAND ${CMAKE_COMMAND} -E env SPHINX_BUILD_CONFIGURATION_FILE=${_docs_config_json} AUTOCONF_JSON_FILE=${AUTOCONF_JSON} VARIANT=${VARIANT} -- sphinx-build -E -b html ${PROJECT_SOURCE_DIR} ${_docs_html_out_dir}
+        BYPRODUCTS ${_docs_html_out_dir}/index.html
+    )
+endmacro()
+
+macro(_spl_create_reports_target)
+    set(_reports_output_dir ${CMAKE_CURRENT_BINARY_DIR}/reports)
+    file(RELATIVE_PATH _rel_reports_output_dir ${PROJECT_SOURCE_DIR} ${_reports_output_dir})
+    set(_reports_html_output_dir ${_reports_output_dir}/html)
+
+    # create the config.json file. This is exported as SPHINX_BUILD_CONFIGURATION_FILE env variable
+    set(_reports_config_json ${_reports_output_dir}/config.json)
+    list(JOIN COMPONENTS_INFO "," _components_info_json)
+    set(_components_info_json "[${_components_info_json}]")
+    # Add the variant specific rst files (e.g, coverage.rst) to the include patterns
+    list(APPEND COMPONENTS_SPHINX_INCLUDE_PATTERNS "${_rel_reports_output_dir}/**")
+    list(JOIN COMPONENTS_SPHINX_INCLUDE_PATTERNS "\",\"" _components_sphinx_include_patterns_json)
+    set(_components_sphinx_include_patterns_json "[\"${_components_sphinx_include_patterns_json}\"]")
+    file(WRITE ${_reports_config_json} "{
+    \"target\": \"reports\",
+    \"include_patterns\": ${_components_sphinx_include_patterns_json},
+    \"reports_output_dir\": \"${_rel_reports_output_dir}\",
+    \"components_info\": ${_components_info_json}
+}")
+
+    # create the variant code coverage rst file
+    set(_coverage_rst ${_reports_output_dir}/coverage.rst)
+    file(WRITE ${_coverage_rst} "
+Code Coverage
+=============
+
+`Report <coverage/index.html>`_
+
+")
+
+    # Iterate over all components and create coverage and doxysphinx variant specific targets
+    foreach(component_info ${COMPONENTS_INFO})

```

Comment:
For every component we need to create specific coverage and doxysphinx targets to make sure the output files are generate in the overall variant sphinx output directory. This will avoid the need to copy all the component coverage and doxygen files from the component directories to the variant directory.